R-package/R/lgb.importance.R

```diff
@@ -75,6 +75,6 @@ lgb.importance <- function(model, percentage = TRUE) {
     )]
   }
 
-  return(tree_imp_dt)
+  return(withVisible(tree_imp_dt)$value)
```

Comment:
I don't think this will have the desired behavior (see https://github.com/microsoft/LightGBM/pull/4194#issuecomment-822023021).

I believe it will do what you want with this suggestion, based on https://stackoverflow.com/a/32989328/3986677

```suggestion
  # adding an empty [] to ensure the table is printed the first time print.data.table() is called
  return(tree_imp_dt[])
```